src/mrpro/operators/functionals/_L2_data_discrepancy.py

```diff
@@ -0,0 +1,38 @@
+"""Squared L2-norm."""
+
+# Copyright 2023 Physikalisch-Technische Bundesanstalt
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#       http://www.apache.org/licenses/LICENSE-2.0
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import torch
+import torch.nn.functional as F
+
+from mrpro.operators import Operator
+
+
+class L2_data_discrepancy(Operator):
+    def __init__(self, data: torch.Tensor) -> None:
+        super().__init__()
+        """Squared L2-norm loss function, i.e.
```

Comment:
I think the descirption does not match what this function does.

It should not divide by the number of elements of it is supposed to be the L2 norm, right?

If I am not mistaken, it returns different values for 1. and 1.+0.j which should also not be the case ..

Reply:
good point. I changed it to mse_data_discrepancy, added remarks (also about the identification of C as R^2) and added some tests